fairlearn/adversarial/_adversarial_mitigation.py

```diff
@@ -0,0 +1,822 @@
+            return
+        if self.backend == "torch" or self.backend == "auto":
+            select = False
+            try:
+                from torch.nn import Module as model
+
+                torch_installed = True
+                if isinstance(
+                    self.predictor_model, (list, model)
+                ) and isinstance(self.adversary_model, (list, model)):
+                    select = True
+                elif self.backend == "torch":
+                    raise ValueError(
+                        _KWARG_ERROR_MESSAGE.format(
+                            "predictor_model and adversary_model",
+                            "a list or torch.nn.Module",
+                        )
+                    )
+            except ImportError:
+                if self.backend == "torch":
+                    raise RuntimeError(_IMPORT_ERROR_MESSAGE.format("torch"))
+            if select:
+                self.backend_ = PytorchEngine
+                return
+        if self.backend == "tensorflow" or self.backend == "auto":
+            select = False
+            try:
+                from tensorflow.keras import Model as model
+
+                tensorflow_installed = True
+                if isinstance(
+                    self.predictor_model, (list, model)
+                ) and isinstance(self.adversary_model, (list, model)):
+                    select = True
+                elif self.backend == "tensorflow":
+                    raise ValueError(
+                        _KWARG_ERROR_MESSAGE.format(
+                            "predictor_model and adversary_model",
+                            "a list or tensorflow.keras.Model",
+                        )
+                    )
+            except ImportError:
+                if self.backend == "tensorflow":
+                    raise RuntimeError(
+                        _IMPORT_ERROR_MESSAGE.format("tensorflow")
+                    )
+            if select:
+                self.backend_ = TensorflowEngine
+                return
+        # The keyword self.backend was weird
+        if self.backend not in ["torch", "tensorflow", "auto"]:
+            raise ValueError(
+                _KWARG_ERROR_MESSAGE.format(
+                    "backend", "one of ['auto', 'torch','tensorflow']"
+                )
+            )
+        # Or no backend is installed
+        if not (torch_installed or tensorflow_installed):
+            raise RuntimeError(
+                _IMPORT_ERROR_MESSAGE.format("torch or tensorflow")
+            )
+        # Or all other cases, a mismatch between model model (and installation)
+        raise ValueError(
+            _KWARG_ERROR_MESSAGE.format(
+                "predictor_model and adversary_model",
+                "a list, torch.nn.Module, or tensorflow.keras.Model. Also, "
+                + "make sure to have installed the corresponding backend",
+            )
+        )
+
+    def _set_predictor_function(self):
```

Comment:
I like this, but even for classifier there are still two options for predictor function (mapping sigmoid to 0/1, or mapping to vector to argmax). Also, how I currently intended it is for the main class AdversarialFairness to be useable in any case, and that stuff like predictor function is inferred (instead of implicitly specifying it through the choice of AdversarialFairness{Classifier,Regressor})

Reply:
Curious what you think. I guess a lot of design choices regarding adversarial fairness could (and maybe should) have been in favor of simpler API and taking away some choices from the user.

Reply:
I generally agree with you, but I think for the user to choose a regressor or a classifier is not something we should hide from them.